src/MCRun.jl

```diff
@@ -181,109 +271,44 @@ save_ham: whether or not to save every energy in a vector, or calculate averages
 save: whether or not to save the parameters and configurations every 1000 steps
 restart: this controls whether to run an equilibration cycle, it additionally requires an integer restartindex which says from which cycle we have restarted the process.
 """
+function ptmc_run!(mc_states, move_strat, mc_params, pot, ensemble, results; save_ham::Bool = false, save::Bool=true, restart::Bool=false,save_dir = pwd())
```

Comment:
- It would be good to be more specific about what/for which purpose something is saved: perhaps change save to save_checkpoint or save_restart
- Is save_ham=true still maintained (it is fine if not, we could perhaps add it to read_save as option)?